File: algos/solutions/a1twopointers/JS/threesum/helper_threesum.js
// helper_threesum.js

function showThreeSumDemo(threeSum) {
    console.log("\n****************************** Three Sum Tests ******************************");
    const testCases = [
        { input: [-1, 0, 1, 2, -1, -4], description: "Example 1" },
        { input: [1, 2, 3, 4, 5], description: "Example 2 (No triplet)" },
        { input: [0, 0, 0, 0], description: "Example 3 (Multiple zeros)" }
    ];

    testCases.forEach(({ input, description }, index) => {
        const result = threeSum(input);
        console.log(`\nTest Case #${index + 1} - ${description}`);
        console.log("-".repeat(80));
        console.log(`Input: [${input.join(", ")}]`);
        console.log(`Output: ${JSON.stringify(result)}`);
        console.log("-".repeat(80));
    });
}

module.exports = { showThreeSumDemo };

File: algos/solutions/a1twopointers/PY/threesum/helper_threesum.py
from threesum.algo import three_sum


def show_three_sum_demo() -> None:
    print("\n****************************** Three Sum Tests ******************************")
    test_cases = [
        {"input": [-1, 0, 1, 2, -1, -4], "description": "Example 1"},
        {"input": [1, 2, 3, 4, 5], "description": "Example 2 (No triplet)"},
        {"input": [0, 0, 0, 0], "description": "Example 3 (Multiple zeros)"}
    ]

    for index, case in enumerate(test_cases, start=1):
        result = three_sum(case["input"])
        print(f"\nTest Case #{index} - {case['description']}")
        print("-" * 80)
        print(f"Input: {case['input']}")
        print(f"Output: {result}")
        print("-" * 80)


if __name__ == "__main__":
    show_three_sum_demo()
